util/linuxfw/iptables_runner.go

```diff
@@ -0,0 +1,485 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build linux
+
+package linuxfw
+
+import (
+	"fmt"
+	"net/netip"
+	"strings"
+
+	"github.com/coreos/go-iptables/iptables"
+	"tailscale.com/net/tsaddr"
+	"tailscale.com/types/logger"
+)
+
+type iptablesInterface interface {
+	// Adding this interface for testing purposes so we can mock out
+	// the iptables library, in reality this is a wrapper to *iptables.IPTables.
+	Insert(table, chain string, pos int, args ...string) error
+	Append(table, chain string, args ...string) error
+	Exists(table, chain string, args ...string) (bool, error)
+	Delete(table, chain string, args ...string) error
+	ClearChain(table, chain string) error
+	NewChain(table, chain string) error
+	DeleteChain(table, chain string) error
+}
+
+type iptablesRunner struct {
+	ipt4 iptablesInterface
+	ipt6 iptablesInterface
+
+	v6Available    bool
+	v6NATAvailable bool
+}
+
+// NewIptablesRunner returns a new iptablesRunner which implements
+// NetfilterRunner interface. It returns an error if the iptables library
+// fails to initialize. The iptablesRunner checks for IPv6 support and
+// IPv6 NAT support on initialization and stores the results. Also stores
+// iptablesInterface for IPv4 and IPv6. In production this interface
+// is a wrapper to *iptables.IPTables.
+func NewIptablesRunner(logf logger.Logf) (*iptablesRunner, error) {
```

Comment:
Thanks for pointing this out. Tho I'm wondering would nftables be Nftables or NFTables since I'm not sure if nf is an abbreviation. Maybe it stands for netfilter?